src/tests/system/tests/test_access_control_ldap_filter.py

```diff
@@ -0,0 +1,251 @@
+from sssd_test_framework.roles.ipa import IPA
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.importance("critical")
+@pytest.mark.require(lambda provider: not isinstance(provider, IPA), "It is not testable with IPA")
+def test_access_filter__single_ldap_attribute_permits_user_login(client: Client, provider: GenericProvider):
+    """
+    :title: LDAP attribute filter permits specific user login
+    :description: Verifies that SSSD allows login only for users matching an specific LDAP attribute.
+    :setup:
+        1. Create users ‘user1’ and ‘user2’
+        2. Configure SSSD with ‘access_provider = ldap|ad’ and ‘*_access_filter = uid|samAccountName = user1’
+        3. Start SSSD
+    :steps:
+        1. Attempt login with `user1`.
+        2. Attempt login with `user2`.
+    :expectedresults:
+        1. `user1` is successfully authenticated.
+        2. `user2` is denied access.
+    :customerscenario: False
+    """
+
+    provider.user("user1").add()
+    provider.user("user2").add()
+
+    if isinstance(provider, (AD, Samba)):
+        client.sssd.domain["access_provider"] = "ad"
+        client.sssd.domain["ad_access_filter"] = "(samAccountName=user1)"
+    else:
+        client.sssd.domain["access_provider"] = "ldap"
+        client.sssd.domain["ldap_access_filter"] = "(uid=user1)"
+
+    client.sssd.start()
+
+    assert client.auth.ssh.password("user1", "Secret123"), "User1 should be able to login!"
+    assert not client.auth.ssh.password("user2", "Secret123"), "User2 should NOT be able to login!"
+
+
+@pytest.mark.topology(KnownTopologyGroup.AnyProvider)
+@pytest.mark.importance("critical")
+@pytest.mark.require(lambda provider: not isinstance(provider, IPA), "It is not testable with IPA")
+@pytest.mark.require(
+    lambda provider: not isinstance(provider, LDAP), "Skipping LDAP until we decide what to do with this test"
+)
+def test_access_filter__group_attributes_permits_user_login(client: Client, provider: GenericProvider):
+    """
+    :title: LDAP attribute permits user login
+    :description:   Tests whether SSSD allows login based on LDAP group membership.
+                    LDAP has options to be tested as rfc23007bis and rfc2307.
+                    The former uses memberOf and the latter member. AD uses memberof, so LDAP will cover member.
+    :setup:
+        1. Create users `user1` and `user2`.
+        2. Create groups `group1` and `group2`.
+        2. Add `user1` to `group1`, and `user2` to `group2`.
+        3. Configure SSSD with `access_provider = ldap|ad` and apply `*_access_filter` to allow only group members.
+        4. Start SSSD.
+    :steps:
+        1. Attempt login with `user1`.
+        2. Attempt login with `user2`.
+    :expectedresults:
+        1. `user1` is successfully authenticated.
+        2. `user2` is denied access.
+    :customerscenario: False
+    """
+
+    provider.group("group1").add().add_member(provider.user("user1").add())
```

Comment:
It would be cleaner to keep instance of the user and use it later

    user1 = provider.user("user1").add()
    provider.group("group1").add().add_member(user1)
    ...
    # later
    assert client.auth.ssh.password(user1.name, "Secret123"), f"{user1.name} should be able to login!"

Reply:
This may be my influence, trying to reduce lines, I'll take the blame regardless. @andreboscatto, @thalman is right, most of the code is like his example, it is much easier to read.

Reply:
Changed :)